packages/php-wasm/compile/Dockerfile

```diff
@@ -950,6 +963,7 @@ RUN source /root/emsdk/emsdk_env.sh && \
     -s ASSERTIONS=0                  \
     -s ERROR_ON_UNDEFINED_SYMBOLS=0  \
     -s NODEJS_CATCH_EXIT=0           \
+	-s NODEJS_CATCH_REJECTION=0      \
```

Comment:
What are the consequences of this?

Reply:
@adamziel 

By default, with the current code, any and all unhandled promise rejections will be sent to a handler registered by emscripten. This handler will then simply throw the error as an exception. 

Unhandled rejections and uncaught exceptions are treated similarly from Node 15 onward, so we don't need to explicitly throw an exception when a promise rejection is unhandled.

The problem is that these (relatively inert) handlers are never cleaned up, so when we create and destroy many php instances in a single process, they stack up. Node JS will throw a warning once there are 10 handlers on this event, and then will not register any more handlers, since it's a suspected memory leak.

Setting that switch to `0` prevents this handler from being added in the first place.

This is chunk of code that is being removed from the emscripten-generated javascript:

https://github.com/WordPress/wordpress-playground/blob/90d6042592cd9037158792c4066db81367a8449b/packages/php-wasm/node/public/php_7_1.js#L116-L118

Reply:
You're right! Sweet:

> As of Node.js 15, the default mode for unhandledRejection is changed to throw (from warn). In throw mode, if an unhandledRejection hook is not set, the unhandledRejection is raised as an uncaught exception

https://nodejs.org/en/blog/release/v15.0.0#throw-on-unhandled-rejections---33021

Reply:
https://github.com/emscripten-core/emscripten/blob/cbf4256d651455abc7b3332f1943d3df0698e990/src/shell.js#L248-L258

Reply:
> Here's the relevant code from Emscripten repo to confirm this is indeed the only effect the flag has:
> 
> https://github.com/emscripten-core/emscripten/blob/cbf4256d651455abc7b3332f1943d3df0698e990/src/shell.js#L248-L258

Drives me insane that they didn't include cleanup logic by default...